source/blog/2018-07-19-ovn-dpdk.html.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Upgraded DPDK support in oVirt
+author: lgoldber
+tags: OVS-DPDK, Open vSwitch, OVN, ansible
+date: 2018-07-19 10:00:00 CET
+comments: true
+published: true
+---
+[DPDK](http://dpdk.org/) (Data Plane Development Kit) provides high-performance packet processing libraries and user space drivers.
+
+[Support for DPDK](https://www.ovirt.org/blog/2017/09/ovs-dpdk/) in oVirt was introduced in 2017, and is now upgraded in terms of both deployment via Ansible and usage via [Open Virtual Network](http://www.ovn.org/).
+
+While still experimental, OVN-DPDK in oVirt is now available in versions 4.2 and upwards.
+
+# What's changed?
+
+### Ansible DPDK host setup
+
+Host configuration for DPDK usage is now automated using Ansible. This primarly includes:
+* Hugepages configuration -- hugepage size and quantity in the kernel.
+* CPU partitioning.
+* Binding NICs to userspace drivers. 
+* OVS-DPDK related configuration (initialization, socket memory, pmd thread core binding, etc).
+
+The role is installed via Ansible galaxy:
+```
+# ansible-galaxy install oVirt.dpdk-setup
+```
+An example playbook:
+```
+- hosts: dpdk_host_0
+  vars:
+    pci_drivers:
+      "0000:02:00.1": "vfio-pci"
+      "0000:02:00.2": "igb"
+      "0000:02:00.3": ""
+    configure_kernel: true
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning). WARNING: When set to `true` it is very likely to trigger reboot of the host, unless all required configuration is already pre-set.
+* `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
+* `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. The empty string represents the default Linux driver for the specified device. Non-DPDK compatible drivers may be set; However, NICs with such drivers will not be configured for OVS-DPDK usage and will not have their CPU isolated in multiple NUMA environments.
```

Comment:
Maybe add that the pci addresses can be obtained using
lshw -class network -businfo

Reply:
Done